lib/ex_doc/formatter/html/templates.ex

```diff
@@ -90,6 +90,24 @@ defmodule ExDoc.Formatter.HTML.Templates do
         group: to_string(group),
         headers: extract_headers(content)
       }
+      |> then(fn item ->
+        case extra do
+          %{search_data: search_data} when is_list(search_data) ->
+            search_data =
+              Enum.map(search_data, fn search_item ->
+                search_item
+                |> Map.put(:id, search_item.title)
```

Comment:
This mapping is admittedly a bit strange, but we massage data primarily designed for the search page into the format expected by sidebar items for autocomplete here (as requested).